C/eval.c

```diff
@@ -657,13 +658,17 @@ typedef struct memBound {
  * Precondition: NULL != dag_bound
  *               dag_node dag[len] and 'dag' is well-typed with 'type_dag'.
  * Postcondition: if the result is 'true'
- *                then 'max(dag_bound->extraCellsBound[0], dag_bound->extraCellsBound[1]) == BOUNDED_MAX'.
+ *                then 'dag_bound->cost == BOUNDED_MAX' or 'dag_bound->const' bounds the dags CPU cost measured in milli weight units
+ *                 and 'max(dag_bound->extraCellsBound[0], dag_bound->extraCellsBound[1]) == BOUNDED_MAX'
  *                     or 'dag_bound->extraCellsBound' characterizes the number of cells needed during evaluation of 'dag'
  *                        and 'dag_bound->extraUWORDBound' characterizes the number of UWORDs needed
  *                              for the frames allocated during evaluation of 'dag'
- *                        and 'dag_bound->extraFrameBound[0]' bounds the the number of stack frames needed during execution of 'dag';
+ *                        and 'dag_bound->extraFrameBound[0]' bounds the the number of stack frames needed during execution of 'dag'.
+ *
+ * :TODO: The cost calculations below are estimated and need to be replaced by experimental data.
  */
 static bool computeEvalTCOBound(memBound *dag_bound, const dag_node* dag, const type* type_dag, const size_t len) {
+  const ubounded overhead = 10;
```

Comment:
In 57f9a933a48fcfd2a9c039a38e33f55cbcfa807c:

I'm a bit confused about this value. It looks like the cost of the 9 core combinators (and witness and disconnect). But aren't these already priced-in as ~half a byte by the fact that they're explicitly encoded in the program in the tx witness?

Also, 10 seems really really high. I guess that the units here are witness bytes, judging by the normalization to "checksig == 50" in the previous commit.

Reply:
Ah, never mind, in the first commit we define the `Weight` type to be in milliweight (and also you say this in the PR description). So this is actually 0.01 weight. Also there is a comment in the Haskell saying "this is just an estimate".

Ok, that answers all my questions.

Though I kinda wish that this hardcoded 10 came as a `#define` in one of the .inc files, to ensure that the Haskell and C are consistent. Fine for now.